enterprise/backend/src/metabase_enterprise/serialization/upsert.clj

```diff
@@ -29,14 +30,14 @@
             [toucan.models :as models]))
 
 (def ^:private identity-condition
-  {Database            [:name]
+  {Database            [:name :engine]
    Table               [:schema :name :db_id]
    Field               [:name :table_id]
    Metric              [:name :table_id]
    Segment             [:name :table_id]
-   Collection          [:name :location]
+   Collection          [:name :location :namespace]
    Dashboard           [:name :collection_id]
-   DashboardCard       [:card_id :dashboard_id :visualiation_settings]
```

Comment:
Maybe we should have some error handling that would warn if these keys don't exist? Either in tests or in runtime checks (that would cause test failures). To prevent this sort of stuff in the future

Reply:
Added a check to `upsert_test.clj` for this.